server/upload.py

```diff
@@ -144,29 +144,39 @@ def __init__(self, dataset, prereviewFolder, csvFile, user, validator):
         # Validation result, created after calling validate()
         self.validationResult = None
 
+    def lines(self, stream):
+        """Generate individual lines of text from a stream."""
+        lastLine = ''
+        try:
+            # Read chunk from stream and split into lines. Always process the
+            # last line with the next chunk, or at the end of the stream,
+            # because it may be incomplete.
+            while True:
+                chunk = lastLine + ''.join(next(stream))
+                lines = chunk.splitlines(True)
```

Comment:
In Python 2 it looks like string functions don't take keyword arguments. It's in Python 3, but I'm not sure in which version it was added.